tool.go

```diff
@@ -121,13 +121,22 @@ func main() {
 
 				for _, pkgPath := range args {
 					pkgPath = filepath.ToSlash(pkgPath)
-					if s.Watcher != nil {
-						s.Watcher.Add(pkgPath)
-					}
 					pkg, err := gbuild.Import(pkgPath, 0, s.InstallSuffix(), options.BuildTags)
 					if err != nil {
 						return err
 					}
+					if s.Watcher != nil {
+						s.Watcher.Add(pkg.Dir)
+
+						// We want to have the package dir watched when calling Import to avoid
+						// races where we could miss a change made and not re-trigger a build.
```

Comment:
Can we turn this around? Put the `if` where it was before, but add `pkg, err := gbuild.Import(pkgPath, 0, s.InstallSuffix(), options.BuildTags)` above the `s.Watcher.Add(pkg.Dir)`.

Also, please try using `gbuild.NewBuildContext(...).Import(...)` with `ImportMode` set to `FindOnly` instead of `gbuild.Import`, cause we're only interested in locating the package.